### ISSUE TITLE
Configuring hop size/frame size

### ISSUE THREAD

**Issue by VariantXYZ** [Rikorose/DeepFilterNet#173]
I’ve been messing around with the plugin on a RPi 3 and it’s functional-ish but it really seems to hammer the CPU. I noticed that the hop_size (and frame size for the model) is set to 10ms, and want to see about playing around with it to see if it can reduce load on the CPU.

I’m not particularly familiar with the model internals, but I’d like to know:
* do you think it would help CPU performance to change the hop and frame size?
* does the model need to be retrained? (i see it as a parameter in the config.ini, so my guess is yes)

**Comment by Rikorose** [comment]
Did you test the deep-filter binary, or the python version? 
A larger hop size and fft size will decrease computational complexity, but if it gets too big, you will hear some smearing artifacts.
Yes the model needs to be retrained. Another option for retraining would be to decrease the network size even further.

**Comment by VariantXYZ** [comment]
Thanks for the quick reply!

I’ve been using the pipewire plugin actually. I believe it’s using the low latency model?

I was thinking about doubling the hop/fft size to see how it would affect CPU utilization. I thought it might be OK just buffering 20ms of audio instead of chunks of 10ms. Thanks for the info about retraining. I’ll look into that.

I’m not actually sure where to start with reducing the network size, but I assumed just increasing the frame size would avoid any loss in quality (though I didn’t think about the smearing).

**Comment by Rikorose** [comment]
> I’ve been using the pipewire plugin actually. I believe it’s using the low latency model?

Yes.

> I was thinking about doubling the hop/fft size to see how it would affect CPU utilization. I thought it might be OK just buffering 20ms of audio instead of chunks of 10ms. Thanks for the info about retraining. I’ll look into that.

Doubling could still be fine, but I would definitely not go bigger than 40 ms window size.

> I’m not actually sure where to start with reducing the network size

You could adjust 3-4 parameters: conv_ch, emb_hidden_dim. df_hidden_dim, and maybe df_num_layers.

**Comment by aaronhsueh0506** [comment]
Hello,

I'm trying to use a dataset with a sample rate of 16000 and NFFT=512 (32ms as a frame) with 50% overlap.
But the model seems to struggle to get the same effect as the sample rate 48k.
Do you have any advice?

**Comment by Rikorose** [comment]
Not really. It can be anything. The data, lr not fitting to your settings or your batch size, etc. You would need to share a lot more infos: config.ini, train log, metric results, used data, example samples etc.

**Comment by aaronhsueh0506** [comment]
Hi,

I have tried some cases.
If I train only stage 1 (ERB gain) it works.
But if I train the whole model, it overfits. (There is some denoising for the first few epochs, but no denoising for the latter.)

Dataset: 48k sample rate, 3s, downsampled to 16k via librosa.resample
FFT size: 512, 50% overlap (so 31.25Hz one bin)
nb_erb: 32, nb_df: 160 (for 5kHz) and df_order = 3.

Because it looks like stage2 is so hard to train that stage1 doesn't work well.
I guess maybe 160*nb_df*2 is too large for the last GroupedFC (GFC_c).

**Comment by Rikorose** [comment]
This sound like underfitting and not overfitting. Try to decrease l2 regularisation.

**Comment by tyrius02** [comment]
@Rikorose Do you have any suggestions on a dataset for retraining? I, too, would like to re-train for a larger hop size (20ms), but it is unclear to me what I should use for the train and test data sets.

Thanks!

**Comment by Rikorose** [comment]
Any dataset that is representable for your use case. I used the DNS4 dataset.

**Comment by VariantXYZ** [comment]
@Rikorose , I've been looking into retraining and was trying to use the English non-emotional DNS4 datasets to keep as closely with your results as possible, but it seems like processing through it all will take a lot of time on my GPU.

Just to make sure, changing hop_size/fft_size and any of the model parameters you mentioned would constitute retraining the model from scratch, right? There's no shortcuts here?

**Comment by Rikorose** [comment]
If you change the hop size or FFT size you need to retrain.
